lib/internal/blob.js

```diff
@@ -188,49 +228,143 @@ class Blob extends JSTransferable {
 
     const span = MathMax(end - start, 0);
 
-    return new InternalBlob(
-      this[kHandle].slice(start, start + span), span, contentType);
+    return createBlob(
+      this[kHandle].slice(start, start + span),
+      span,
+      contentType);
   }
 
-  async arrayBuffer() {
+  /**
+   * @returns {Promise<ArrayBuffer>}
+   */
+  arrayBuffer() {
+    if (!isBlob(this))
+      return PromiseReject(new ERR_INVALID_THIS('Blob'));
+
+    // If there's already a promise in flight for the content,
+    // reuse it, but only once. After the cached promise resolves
+    // it will be cleared, allowing it to be garbage collected
+    // as soon as possible.
+    if (this[kArrayBufferPromise])
+      return this[kArrayBufferPromise];
+
     const job = new FixedSizeBlobCopyJob(this[kHandle]);
 
     const ret = job.run();
+
+    // If the job returns a value immediately, the ArrayBuffer
+    // was generated synchronously and should just be returned
+    // directly.
     if (ret !== undefined)
       return PromiseResolve(ret);
 
     const {
       promise,
       resolve,
-      reject
+      reject,
     } = createDeferredPromise();
+
     job.ondone = (err, ab) => {
       if (err !== undefined)
         return reject(new AbortError());
       resolve(ab);
     };
+    this[kArrayBufferPromise] =
+    SafePromisePrototypeFinally(
+      promise,
+      () => this[kArrayBufferPromise] = undefined);
 
-    return promise;
+    return this[kArrayBufferPromise];
   }
 
+  /**
+   * @returns {Promise<string>}
+   */
   async text() {
+    if (!isBlob(this))
+      throw new ERR_INVALID_THIS('Blob');
+
     const dec = new TextDecoder();
     return dec.decode(await this.arrayBuffer());
   }
+
+  /**
+   * @returns {ReadableStream}
+   */
+  stream() {
+    if (!isBlob(this))
+      throw new ERR_INVALID_THIS('Blob');
+
+    const self = this;
+    return new lazyReadableStream({
+      async start(controller) {
+        const ab = await self.arrayBuffer();
+        controller.enqueue(new Uint8Array(ab));
+        controller.close();
+      }
+    });
+  }
+}
+
+function ClonedBlob() {
+  return makeTransferable(ReflectConstruct(function() {}, [], Blob));
+}
+ClonedBlob.prototype[kDeserialize] = () => {};
+
+function createBlob(handle, length, type = '') {
+  return makeTransferable(ReflectConstruct(function() {
+    this[kHandle] = handle;
+    this[kType] = type;
+    this[kLength] = length;
+  }, [], Blob));
 }
 
 ObjectDefineProperty(Blob.prototype, SymbolToStringTag, {
   configurable: true,
   value: 'Blob',
 });
 
-InternalBlob.prototype.constructor = Blob;
-ObjectSetPrototypeOf(
-  InternalBlob.prototype,
-  Blob.prototype);
+function resolveObjectURL(url) {
+  url = `${url}`;
+  try {
+    const parsed = new lazyURL(url);
+
+    const split = StringPrototypeSplit(parsed.pathname, ':');
```

Comment:
Wouldn't it be more performant to specify how many chunks we are expecting?
```suggestion
    const split = StringPrototypeSplit(parsed.pathname, ':', 2);
```